test/smoke/framework/testCore/src/main/java/com/microsoft/applicationinsights/smoketest/AiSmokeTest.java

```diff
@@ -225,19 +236,38 @@ public void run() {
 	}
 
 	@ClassRule
-	public static TestWatcher agentModeLoader = new TestWatcher() {
+	public static TestWatcher classOptionsLoader = new TestWatcher() {
 		@Override
 		protected void starting(Description description) {
+			System.out.println("Configuring test class...");
 			UseAgent ua = description.getAnnotation(UseAgent.class);
 			if (ua != null) {
 				agentMode = ua.value();
 				System.out.println("AGENT MODE: "+agentMode);
 			}
+			WithDependencyContainers wdc = description.getAnnotation(WithDependencyContainers.class);
+			if (wdc != null) {
+				for (DependencyContainer container : wdc.value()) {
```

Comment:
@littleaj Can we do this in parallel? How about using a thread pool to add it in a ConcurrentList? If we have many dependencies in future this should help speeding up according to me.